src/main/scala/amailp/intellij/robot/psi/reference/PythonKeywordToDefinitionReference.scala

```diff
@@ -31,41 +30,37 @@ class PythonKeywordToDefinitionReference(element: PsiElement, textRange: TextRan
   override def resolve: PsiElement = multiResolve(false).headOption.map(_.getElement).orNull
 
   def findMatchingInLibraries(reference: String) = {
-    val referenceRegex = Pattern.quote(reference.replaceAll(" ", "").toLowerCase).r
     val libraryQNames = currentRobotFile.getImportedLibraries.map(l => QualifiedName.fromDottedString(l.getText))
 
     type find[T] = (String, Project, Boolean) => util.Collection[T]
 
     def mmm(qName: QualifiedName, pySomething: NavigationItem) =
       qName.getComponentCount == 1 || (pySomething.getPresentation.getLocationString contains qName.removeLastComponent().toString)
 
-    def findWith[T<:NavigationItem](index: find[T], find: (T, Regex) => Iterable[PsiElement]) = for {
-        qName <- libraryQNames
-        elem <- index(qName.getLastComponent, element.getProject, true)
-        if mmm(qName, elem)
-        psiElement <- find(elem, referenceRegex)
-      } yield psiElement
+    def findWith[T <: NavigationItem](index: find[T], find: (T, String) => Iterable[PsiElement]) = for {
+      qName <- libraryQNames
+      elem <- index(qName.getLastComponent, element.getProject, true)
+      if mmm(qName, elem)
+      psiElement <- find(elem, reference)
+    } yield psiElement
 
     findWith(PyModuleNameIndex.find, findInPyFile) ++ findWith(PyClassNameIndex.find, findInPyClass)
   }
 
-  def findInPyFile(pyFile: PyFile, reference: Regex) = for {
-      keyword <- PsiTreeUtil.findChildrenOfType(pyFile, classOf[PyFunction])
-      if pyFunctionMatches(keyword, reference) && Option(keyword.getContainingClass).isEmpty
-    } yield keyword
-
-  def findInPyClass(pyClass: PyClass, reference: Regex) = for {
-      pyClass <- pyClass +: pyClass.getSuperClasses(null)
-      keyword <- pyClass.getMethods
-      if pyFunctionMatches(keyword, reference)
-    } yield keyword
-
-  def pyFunctionMatches(pyFunc: PyFunction, reference: Regex) : Boolean = {
-    val functionName = pyFunc.getName
-    !functionName.startsWith("_") && (functionName.replaceAll("_", "").toLowerCase match {
-      case reference() => true
-      case _ => false
-    })
+  def findInPyFile(pyFile: PyFile, reference: String) = for {
+    keyword <- PsiTreeUtil.findChildrenOfType(pyFile, classOf[PyFunction])
+    if Option(keyword.getContainingClass).isEmpty && pyElementMatches(keyword, reference)
+  } yield keyword
+
+  def findInPyClass(pyClass: PyClass, reference: String) = for {
+    keyword <- List.apply(findInPyClassByName(reference, pyClass))
+    if keyword != null
+  } yield keyword
+
+  def findInPyClassByName(name: String, pyClass: PyClass): PyFunction = {
+    val process: MethodFinderWithoutUnderscoresAndSpaces = new MethodFinderWithoutUnderscoresAndSpaces(name)
```

Comment:
[minor] remove redundant type here. It is clear from the new in the right hand side.